### PR TITLE
Keep tags based on regex

### PR DESCRIPTION
**Pull request by dmaes**
Adds a config option that takes a regex. Tags matching this regex will not be purged.

### CODE REVIEW

**Comment by roman-vynar** [comment]
Thanks for using docker-registry-ui!

By default, purge_tags_keep_regexp is empty. Empty regex is matched as true against any string.
This means that if you don't set purge_tags_keep_regexp we won't have anything purged at all.
Please add a check, if regex is empty - don't even match the regex.

**Comment by dmaes** [comment]
Forgot about empty regex indeed

File: registry/tasks.go [hunk 1]
```diff
@@ -83,10 +84,15 @@ func PurgeOldTags(client *Client, purgeDryRun bool, purgeTagsKeepDays, purgeTags
 		sort.Sort(sortedTags)
 		repos[repo] = sortedTags
 
-		// Filter out tags by retention days.
+		// Filter out tags by retention days and regexp
 		for _, tag := range repos[repo] {
+			regexpKeep := false
+			if purgeTagsKeepRegexp != "" {
+				regexpMatch, _ := regexp.MatchString(purgeTagsKeepRegexp, tag.name)
+				regexpKeep = regexpMatch
```
Comment (roman-vynar) [inline registry/tasks.go:92]:
```
				regexpMatch, _ := regexp.MatchString(purgeTagsKeepRegexp, tag.name)
				regexpKeep = regexpMatch
```
can be simplified to
```
				regexpKeep, _ = regexp.MatchString(purgeTagsKeepRegexp, tag.name)
```

Comment (dmaes) [reply registry/tasks.go]: Not much go experience, wasn't sure that that worked with `=` too or that `something, _` always had to be `:=`

**Pull request merged**